pallets/identity/src/claims.rs

```diff
@@ -452,11 +452,10 @@ impl<T: Config> Module<T> {
     ) -> DispatchResult {
         let (pk, sk) = Self::get_claim_keys(target, claim_type, issuer, scope);
         // Ensure that claim exists
-        let investor_unique_scope_id = match Claims::try_get(&pk, &sk)
-            .map_err(|_| Error::<T>::ClaimDoesNotExist)?
-            .claim
-        {
-            Claim::InvestorUniqueness(_, scope_id, _) => Some(scope_id),
+        let claim = Claims::take(&pk, &sk).ok_or(Error::<T>::ClaimDoesNotExist)?;
```

Comment:
I'm a bit concerned updating the storage here, since it can fail later.  The caller of this function (`revoke_systematic_cdd_claims`) for some reason doesn't forward the error. I haven't analyzed why, though.

Reply:
Good catch.  In that cause the claim type is always a CDD claim, which wouldn't trigger the other two errors below.  I have created a ticket for `revoke_systematic_cdd_claims`, since we shouldn't be ignoring errors like that.

I will change this code anyways.